GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+}
+```
+
+### Proposals
+
+`Proposals` are item to be voted on. They can be submitted by any Atom holder via a `TxSubmitProposal` transaction.
+
+```Go
+type TxSubmitProposal struct {
+  Title           string        //  Title of the proposal
+  Description     string        //  Description of the proposal
+  Type            string        //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category        bool          //  false=regular, true=urgent
+  InitialDeposit  uint64        //  Initial deposit paid by sender. Must be strictly positive.
+}
+
+type Proposal struct {
+  Title             string              //  Title of the proposal
+  Description       string              //  Description of the proposal
+  Type              string              //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category          bool                //  false=regular, true=urgent
+  Deposit           uint64              //  Current deposit on this proposal. Initial value is set at InitialDeposit
+  SubmitBlock       uint64              //  Height of the block where TxSubmitProposal was included
+  VotingStartBlock  uint64              //  Height of the block where MinDeposit was reached. -1 if MinDeposit is not reached.
+  Votes             map[string]uint64   //  Votes for each option (Yes, No, NoWithVeto, Abstain)
+}
+```
+
+Each `Proposal` is identified by its unique `proposalID`. 
+
+Additionaly, four lists will be linked to each proposal:
+- `DepositorList`: List of addresses that deposited on the proposal with their associated deposit
+- `VotersList`: List of addresses that voted **under each validator** with their associated option
+- `InitVotingPowerList`: Snapshot of validators' voting power **when proposal enters voting period** (only saves validators whose voting power is >0).
+- `MinusesList`: List of minuses for each validator. Used to compute validators' voting power when they cast a vote.
+
+Two final parameters, `InitTotalVotingPower` and `InitProcedureNumber` associated with `proposalID` will be saved when proposal enters voting period.
+
+We also introduce `ProposalProcessingQueue` which lists all the `ProposalIDs` of proposals that reached `MinDeposit` from oldest to newest. Each round, the oldest element of `ProposalProcessingQueue` is checked during `BeginBlock` to see if `CurrentBlock == VotingStartBlock + InitProcedure.VotingPeriod`. If it is, then the application checks if validators in `InitVotingPowerList` have voted and, if not, applies `GovernancePenalty`. After that proposal is ejected from `ProposalProcessingQueue` and the new first element of the queue is evaluated. Note that if a proposal is urgent and accepted under the special condition, its `ProposalID` must be ejected from `ProposalProcessingQueue`.
+
+A `TxSubmitProposal` transaction can be handled according to the following pseudocode
+
+```
+// PSEUDOCODE //
```

Comment:
I think we'd be better not including the pseudo code in the spec (still keep a copy though!) but just a short description of any of the programmatic nuances should be sufficient... This comment applies to all the pseudocode sections

Reply:
Agreed. I'll leave it until @sunnya97 reviews it and then remove it in a later commit.